pkg/api/v1beta1/types.go

```diff
@@ -747,6 +747,27 @@ type ReplicationController struct {
 	Labels       map[string]string          `json:"labels,omitempty" description:"map of string keys and values that can be used to organize and categorize replication controllers"`
 }
 
+// JobState is the state of a job, either input (create, update) or as output (list, get).
+type JobState struct {
+	Completions int               `json:"replicas" description:"number of replicas (desired or observed, as appropriate)"`
+	JobSelector map[string]string `json:"jobSelector,omitempty" description:"label keys and values that must match in order to be controlled by this job"`
+	PodTemplate PodTemplate       `json:"podTemplate,omitempty" description:"template for pods to be created by this job when the observed number of replicas is less than the desired number of replicas"`
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+	TypeMeta `json:",inline"`
+	Items    []Job `json:"items" description:"list of jobs"`
+}
+
+// Job represents the configuration of a job.
+type Job struct {
+	TypeMeta     `json:",inline"`
+	DesiredState JobState          `json:"desiredState,omitempty" description:"specification of the desired state of the job"`
+	CurrentState JobState          `json:"currentState,omitempty" description:"current state of the job; populated by the system, read-only"`
```

Comment:
I'm doing it this way to follow a pattern. Unsure why v1beta1/2 did it this way for other resources, it's fixed in 3 though. Don't expect it to be much of a burden, but if you have reasons I can think about deviating.